jax/_src/numpy/reductions.py

```diff
@@ -65,6 +65,20 @@ def _upcast_f16(dtype: DTypeLike) -> DType:
     return np.dtype('float32')
   return np.dtype(dtype)
 
+def _promote_integer_dtype(dtype: DTypeLike) -> DTypeLike:
```

Comment:
I have it as a small private utility for only `_reductions` and `_make_cumulative_reduction` -- I don't think it should really be used anywhere else, since this behavior only applies to accumulators

Reply:
Sounds good - sorry for all the churn/bikeshedding on this